packages/core/react/src/WalletProviderBase.tsx

```diff
@@ -8,54 +8,58 @@ import type {
 } from '@solana/wallet-adapter-base';
 import { WalletNotConnectedError, WalletNotReadyError, WalletReadyState } from '@solana/wallet-adapter-base';
 import type { PublicKey } from '@solana/web3.js';
-import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import React, { type ReactNode, useCallback, useEffect, useMemo, useRef, useState } from 'react';
 import { WalletNotSelectedError } from './errors.js';
 import { WalletContext } from './useWallet.js';
-import type { WalletProviderProps } from './WalletProvider.js';
 
-type Props = Readonly<
-    Omit<WalletProviderProps, 'autoConnect' | 'localStorageKey'> & {
-        adapter: Adapter | null;
-        isUnloadingRef: React.RefObject<boolean>;
-        // NOTE: The presence/absence of this handler implies that auto-connect is enabled/disabled.
-        onAutoConnectRequest?: () => Promise<void>;
-        onConnectError: () => void;
-        onSelectWallet: (walletName: WalletName | null) => void;
-    }
->;
+export interface WalletProviderBaseProps {
+    children: ReactNode;
+    wallets: Adapter[];
+    adapter: Adapter | null;
+    isUnloadingRef: React.RefObject<boolean>;
+    // NOTE: The presence/absence of this handler implies that auto-connect is enabled/disabled.
+    onAutoConnectRequest?: () => Promise<void>;
+    onConnectError: () => void;
+    onError?: (error: WalletError, adapter?: Adapter) => void;
+    onSelectWallet: (walletName: WalletName | null) => void;
+}
 
 export function WalletProviderBase({
-    adapter,
     children,
+    wallets: adapters,
+    adapter,
     isUnloadingRef,
     onAutoConnectRequest,
     onConnectError,
-    onError,
+    onError = (error, adapter) => {
+        console.error(error, adapter);
+        if (error instanceof WalletNotReadyError && typeof window !== 'undefined' && adapter) {
+            window.open(adapter.url, '_blank');
+        }
+    },
     onSelectWallet,
-    wallets: adapters,
-}: Props) {
-    const isConnecting = useRef(false);
+}: WalletProviderBaseProps) {
+    const isConnectingRef = useRef(false);
     const [connecting, setConnecting] = useState(false);
-    const isDisconnecting = useRef(false);
+    const isDisconnectingRef = useRef(false);
     const [disconnecting, setDisconnecting] = useState(false);
     const [publicKey, setPublicKey] = useState(() => adapter?.publicKey ?? null);
     const [connected, setConnected] = useState(() => adapter?.connected ?? false);
-    const handleError = useCallback(
-        (error: WalletError, adapter?: Adapter) => {
+
+    const handleErrorRef = useRef((error: WalletError, adapter?: Adapter) => {
+        if (!isUnloadingRef.current) {
+            onError(error, adapter);
+        }
+        return error;
+    });
+    useEffect(() => {
+        handleErrorRef.current = (error, adapter) => {
             if (!isUnloadingRef.current) {
-                if (onError) {
-                    onError(error, adapter);
```

Comment:
If you make the call to `handleErrorRef` here, all of the rest of the code gets way simpler.

Reply:
Good point, I wasn't happy with how I did this.